dlib/image_transforms/interpolation.h

```diff
@@ -965,6 +965,46 @@ namespace dlib
         swap(img, temp);
     }
 
+// ----------------------------------------------------------------------------------------
+
+    template <typename image_type>
+    point_transform_affine letterbox_image (
+        const image_type& img_in,
+        image_type& img_out,
+        long size
+    )
+    {
+        DLIB_CASSERT(size > 0, "size must be bigger than zero, but was " << size);
+        // the scaling factor of the image
+        const auto scale = size / std::max<double>(img_in.nr(), img_in.nc());
+
+        // early return if the image has already the requested size and no padding is needed
+        if (scale == 1 and img_in.nr() == img_in.nc())
```

Comment:
Ha, `and`.  :)

Use `&&` instead of `and` though.  `&&` is way more idiomatic in C++ than `and`.  I'm pretty sure like 95% of C++ developers don't even know you can use `and`.

Reply:
Yes, I usually use `and` in my code, I know it's not that idiomatic, but I kind of like it :P.